### PR TITLE
Feat/sort by time

### PR DESCRIPTION
**Pull request by viru2001**
- user can sort note by time 
    -  Newest First
    - Oldest First

sort by time (1)

### CODE REVIEW

File: src/frontend/reducer/notesReducer.js [hunk 1]
```diff
@@ -88,6 +89,16 @@ const notesReducer = (notesState, { type, payload }) => {
           note._id === payload._id ? payload : note
```
Comment (MonikaJhingan) [inline src/frontend/reducer/notesReducer.js:89]: Use of payload nicely done.

**Review by anuj-kumary — APPROVED** [review]
Great naming convention. Codes are readable and clean. Great work.

File: src/frontend/components/Sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -1,52 +1,117 @@
-import { NavLink } from "react-router-dom";
+import { useNotes } from "frontend/context";
+import { NavLink, useLocation } from "react-router-dom";
 import "./Sidebar.css";
 const Sidebar = () => {
   const activeStyle = {
     backgroundColor: "#a7f6ff",
```
Comment (MonikaJhingan) [inline src/frontend/components/Sidebar/Sidebar.jsx:6]: Use of var is more preferrable as compared to  directly writing about the colors.

File: src/frontend/utils/sortUtils/sortNotes.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    return [...notes].sort(
+      (a, b) => new Date(a.createdAt) - new Date(b.createdAt)
+    );
+  if (sortBy === "NEWEST_FIRST")
+    return [...notes].sort(
+      (a, b) => new Date(b.createdAt) - new Date(a.createdAt)
+    );
+  return notes;
+};
+export { sortNotes };
```
Comment (MonikaJhingan) [inline src/frontend/utils/sortUtils/sortNotes.js:12]: The logic here is well implemented and the code quality is good.

File: src/frontend/reducer/NotesReducer.js [hunk 4]
```diff
@@ -88,6 +89,16 @@ const notesReducer = (notesState, { type, payload }) => {
...
+        sortByTime:
+          payload === "oldest-first"
+            ? "OLDEST_FIRST"
+            : payload === "newest-first"
+            ? "NEWEST_FIRST"
+            : "",
+      };
     default:
       throw new Error("Unhandled action type");
   }
```
Comment (MonikaJhingan) [inline src/frontend/reducer/NotesReducer.js:104]: Exception handling is done perfectly as it will help in passing test cases.

**Review by MonikaJhingan — COMMENTED** [review]
Cn be merged.

**Pull request merged**